### DISCUSSION TITLE
PLCOpen XML HowTo

### DISCUSSION THREAD

**Discussion by marwern** [jubnzv/iec-checker#11] [Q&A]
Hi, I installed v0.3 on Win10 and tried to run test/plcopen/example.xml. I get "ParserError:" while mentioned st example works fine. Furthermore, --help command throws a uncaught exception.

I had a look at *.ml file but for me as non-expert it seems nothing more is missing. Any ideas?

Thank you.

**Comment by jubnzv** [comment]
Hi @marwern, thank you for the report.

Could you please provide the complete commands with their output?

I personally don't use Windows, but I'll take a look ASAP.

**Reply by marwern** [reply]
Hi, thanks for the instant feedback. Below are a few examples. No stress. I really like your project and it seems to be quite helpful for ST programs.

Good (ST):
```
C:\<localPath>\iec_checker>iec_checker_Windows_x86_64.exe "C:\< localPath>\iec_checker\iec-checker-master\test\st\dead-code.st"

20:7 UnreachableCode: Code block will never be reached

25:7 UnreachableCode: Code block will never be reached

9:10 UnreachableCode: Code block will never be reached

15:5 UnusedVariable: Found unused local variable: A

17:6 PLCOPEN-L17: Each IF instruction should have an ELSE clause
```

Bad (PLCOpen):
```
C:\<localPath>\iec_checker>iec_checker_Windows_x86_64.exe "C:\<localPath>\iec_checker\iec-checker-master\test\plcopen\example.xml"

1:1 ParserError:
```


Bad (--help):
```
C:\<localPath>\iec_checker>iec_checker_Windows_x86_64.exe "C:\<localPath>\iec_checker\iec-checker-master\test\st\dead-code.st" --help

Uncaught exception:


  Not_found
```


Bad (--help):
```
C:\<localPath>\iec_checker>iec_checker_Windows_x86_64.exe --help

Uncaught exception:


  Not_found
```

**Reply by jubnzv** [reply]
> Bad (PLCOpen):

The correct command would be:
```
iec_checker_Windows_x86_64.exe -i xml "C:\<localPath>\iec_checker\iec-checker-master\test\plcopen\example.xml"
```
We intentionally do not try to pick up the correct input format based on the file name, because there may be ambiguities. For example, `.xml` file type is used for both PLCOpen XML and Schweitzer Engineering Laboratories XML.

> Bad (--help):

Got it, thanks. I'll take a look. There's something wrong on Windows here.

**Reply by jubnzv** [reply]
I fixed an error with `--help` and released a new version: https://github.com/jubnzv/iec-checker/releases/tag/v0.4

The problem is in the third-party dependency: https://github.com/rbardou/clap/issues/1.

It occurs only when running `iec-checker` using Powershell or `cmd.exe`. In Cygwin everything works well.

This has not been fixed yet, so when building IECChecker from source code on Windows, the same problem will occur. This is required to build a [clap](https://github.com/rbardou/clap/) library manually from source first, and then to build IECChecker itself. But the binary from the release is already built with a fixed version of `clap`.

**Reply by marwern** [reply]
Thank you very much, @jubnzv
